components/playlist/mojom/playlist.mojom

```diff
@@ -39,24 +39,34 @@ struct PlaylistItem {
   int8 last_played_position;
 };
 
-// Used by the WebUI page to bootstrap bidirectional communication.
+// Used by the UI to bootstrap bidirectional communication.
 interface PageHandlerFactory {
-  // The WebUI page's |BrowserProxy| singleton calls this method when the page
-  // is first initialized.
-  CreatePageHandler(pending_remote<Page> page,
-                    pending_receiver<PageHandler> handler);
+  // On Desktop, the WebUI page's |BrowserProxy| singleton calls this method 
+  // when the page is first initialized.
+  BindInterface(pending_remote<ServiceObserver> service_observer,
+                pending_receiver<Service> service);
 };
 
-// Browser-side handler for requests from WebUI page.
-interface PageHandler {
+// Service-side handler for requests from UI.
+interface Service {
+  // Getters for playlists.
   GetAllPlaylists() => (array<Playlist> playlists);
-
   GetPlaylist(string id) => (Playlist? playlist);
 
+  // Getters for playlist items.
+  GetAllPlaylistItems() => (array<PlaylistItem> items);
+  GetPlaylistItem(string id) => (PlaylistItem? item);
+
+  // Store all media files from given URL.
   AddMediaFilesFromPageToPlaylist(string playlist_id, url.mojom.Url url);
   
   // Store all media files from tabs in this window.
   AddMediaFilesFromOpenTabsToPlaylist(string playlist_id);
+  AddMediaFilesFromActiveTabToPlaylist(string playlist_id);
```

Comment:
@sangwoo108 i think we are missing few functions from playlist_service. should we handle it in follow up PR ?

Reply:
Yes. AFAIK, item re-odering is not ready now. Is there anything else I'm missing?

Reply:
I can think of a couple of functions but all good for this PR. I will start adding functions in follow up changes.